src/analysis/special_functions/bernstein.lean

```diff
@@ -0,0 +1,301 @@
+The Bernstein polynomials, as continuous functions on `[0,1]`.
+-/
+def bernstein (n ν : ℕ) : C(I, ℝ) :=
+(bernstein_polynomial ℝ n ν).as_continuous_map_on I
+
+@[simp] lemma bernstein_apply (n ν : ℕ) (x : I) :
+  bernstein n ν x = n.choose ν * x^ν * (1-x)^(n-ν) :=
+begin
+  dsimp [bernstein, polynomial.as_continuous_map_on, polynomial.as_continuous_map,
+    bernstein_polynomial],
+  simp,
+end
+
+lemma bernstein_nonneg {n ν : ℕ} {x : I} :
+  0 ≤ bernstein n ν x :=
+begin
+  simp only [bernstein_apply],
+  exact mul_nonneg
+    (mul_nonneg (nat.cast_nonneg _) (pow_nonneg (by unit_interval) _))
+    (pow_nonneg (by unit_interval) _),
+end
+
+/-!
+We now give a slight reformulation of `bernstein_polynomial.variance`.
+-/
+
+namespace bernstein
+
+/--
+Send `k : fin (n+1)` to the equally spaced points `k/n` in the unit interval.
+-/
+def z {n : ℕ} (k : fin (n+1)) : I :=
+⟨(k : ℝ) / n,
+  begin
+    cases n,
+    { norm_num },
+    { have h₁ : 0 < (n.succ : ℝ) := by exact_mod_cast (nat.succ_pos _),
+      have h₂ : ↑k ≤ n.succ := by exact_mod_cast (fin.le_last k),
+      rw [set.mem_Icc, le_div_iff h₁, div_le_iff h₁],
+      norm_cast,
+      simp [h₂], },
+  end⟩
+
+local postfix `/ₙ`:90 := z
+
+lemma probability (n : ℕ) (x : I) :
+  ∑ k : fin (n+1), bernstein n k x = 1 :=
+begin
+  have := bernstein_polynomial.sum ℝ n,
+  apply_fun (λ p, polynomial.aeval (x : ℝ) p) at this,
+  simp [alg_hom.map_sum, finset.sum_range] at this,
+  exact this,
+end
+
+lemma variance {n : ℕ} (h : 0 < (n : ℝ)) (x : I) :
+  ∑ k : fin (n+1), (x - k/ₙ : ℝ)^2 * bernstein n k x = x * (1-x) / n :=
+begin
+  have h' : (n : ℝ) ≠ 0 := (ne_of_lt h).symm,
+  apply_fun (λ x : ℝ, x * n) using group_with_zero.mul_right_injective h',
+  apply_fun (λ x : ℝ, x * n) using group_with_zero.mul_right_injective h',
+  dsimp,
+  conv_lhs { simp only [finset.sum_mul, z], },
+  conv_rhs { rw div_mul_cancel _ h', },
+  have := bernstein_polynomial.variance ℝ n,
+  apply_fun (λ p, polynomial.aeval (x : ℝ) p) at this,
+  simp [alg_hom.map_sum, finset.sum_range, ←polynomial.nat_cast_mul] at this,
+  convert this using 1,
+  { congr' 1, funext k,
+    rw [mul_comm _ (n : ℝ), mul_comm _ (n : ℝ), ←mul_assoc, ←mul_assoc],
+    congr' 1,
+    field_simp [h],
+    ring, },
+  { ring, },
+end
+
+end bernstein
+
+open bernstein
+
+local postfix `/ₙ`:2000 := z
+local notation `|`x`|` := abs x
+
+/--
+The `n`-th approximation of a continuous function on `[0,1]` by Bernstein polynomials,
+given by `∑ k, f (k/n) * bernstein n k x`.
+-/
+def bernstein_approximation (n : ℕ) (f : C(I, ℝ)) : C(I, ℝ) :=
+∑ k : fin (n+1), f k/ₙ • bernstein n k
+
+/-!
+We now set up some of the basic machinery of the proof that the Bernstein approximations
+converge uniformly.
+
+A key player is the set `S f ε h n x`,
+for some function `f : C(I, ℝ)`, `h : 0 < ε`, `n : ℕ` and `x : I`.
+This is the set of points `k` in `fin (n+1)` such that `f(k/n)` is within `ε/2` of `f x`.
+
+We show that if `k ∉ S`, then `1 ≤ δ^-2 * (x - k/n)^2`.
```

Comment:
you should say what \delta is here. Maybe even introduce an abbreviation `\delta := f.modulus (ε/2) (half_pos h)` for the remainder of the file?

Reply:
Ah, yes. I've introduced it earlier now.